guided-match/guided-match-service.yaml

```diff
@@ -487,6 +487,26 @@ components:
           description: Ordered collection of defined answers (for radio and checkbox types)
           items:
             $ref: '#/components/schemas/AnswerDefinition'
+        failureValidations:
+          type: array
+          description: "Array of validations to be performed on the provided answer and corresponding error messages"
+          items:
+            $ref: '#/components/schemas/FailureValidation'
+            
+    FailureValidation:
+      description: "A validation check to be performed on an answer to a question and error message if it fails"
+      type: object
+      properties:
+        failureValidationTypeCode:

```

Comment:
Should this be an enumeration of values i.e. `noSelection`, `noValue` etc as specified in the HLD - therefore auto-documented and extensible as new question / failure types come into use?

Reply:
I'd rather not make this a enum as it isn't a limited list, it will expand with more validations as time goes on.  I see your point about it being self-documenting but that'd only be if the 'description' is updated to explain what the entries in the enum mean. The items in the HLD are intended to be examples - they represent what we know so far. I expect that more will need to be added.  I sort of see the point that the UI needs to be updated for a new validation so why not the API as well, but in theory the API should be approved by governance whereas a UI update is just part of the SDLC.
How strongly do you feel about this?